### PR TITLE
Convert IqApplication sub-classes to Java so that pipeline elements are found.  CLM-10570

### PR DESCRIPTION
**Pull request by tneer**
https://jenkins.zion.aws.s/blue/organizations/jenkins/integrations%2Fjenkins%2Fsonatype-nexus-platform-plugin-feature/activity?branch=CLM-10570_pipeline_syntax
https://issues.sonatype.org/browse/CLM-10570

The build won't pass until we can release the nexus-platform-api

### CODE REVIEW

File: src/test/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorDescriptorTest.groovy [hunk 1]
```diff
@@ -146,27 +148,37 @@ abstract class IqPolicyEvaluatorDescriptorTest
...
+      GlobalNexusConfiguration.globalNexusConfiguration.iqConfigs = [new NxiqConfiguration('http://server/url',
+          null)]
       def job = Mock(Job)
 
     when:
       descriptor.doFillApplicationIdItems('credentialsId', job)
 
     then:
-      1 * IqUtil.doFillIqApplicationItems('credentialsId', job)
+      // this is a bit odd, but because we're mixing Groovy and Java classes SelectedApplication, we can't
```
Comment (whyjustin) [inline src/test/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorDescriptorTest.groovy]: Probably nice to have a factory for the IqUtil to facilitate this kind of thing (nothing needed now though).

File: src/test/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorDescriptorTest.groovy [hunk 2]
```diff
@@ -146,27 +148,37 @@ abstract class IqPolicyEvaluatorDescriptorTest
...
+      // use GroovyMock to verify interactions.  What we are testing here is that IqUtil.doFillIqApplicationItems
+      // was invoked with 'credentialsId'.  Since the Jenkins credential hasn't been created
+      // IqUtil.doFillIqApplicationItems will throw throw an exception, and then we can check the exception message
```
Comment (DennisWarren) [inline src/test/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorDescriptorTest.groovy]: 'throw throw an exception' typo

File: src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorWorkflowStep.groovy [hunk 3]
```diff
@@ -76,6 +77,9 @@ class IqPolicyEvaluatorWorkflowStep
     else if (iqApplication instanceof String) {
       this.iqApplication = new SelectedApplication(iqApplication)
     }
+    else if (iqApplication instanceof UninstantiatedDescribable) {
+      this.iqApplication = ((UninstantiatedDescribable) iqApplication).instantiate()
```
Comment (DennisWarren) [inline src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorWorkflowStep.groovy]: not sure I follow what's going on here.

Comment (tneer) [reply src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorWorkflowStep.groovy]: >not sure I follow what's going on here.
Edit: In particular, when does this become an UninstatiatedDescribable?

I'm not going to claim that I understand it.  This is what the debugger told me was actually being passed in and why it was skipping to the throw exception block.   Jenkins uses UninstantiatedDescribable to convert from pipeline code to actual objects.   Not much documentation on it that I have found.  

https://github.com/jenkinsci/structs-plugin/blob/master/plugin/src/main/java/org/jenkinsci/plugins/structs/describable/UninstantiatedDescribable.java

File: src/test/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorDescriptorTest.groovy (hunk 2 above)
Comment (tneer) [reply src/test/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorDescriptorTest.groovy]: > 'throw throw an exception' typo

Thank you, thank you.  Fixed in 9bc533c10f56647c85983a058fa996ebffc1ebaf

File: src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorWorkflowStep.groovy (hunk 3 above)
Comment (fjmilens3) [reply src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorWorkflowStep.groovy]: > This is what the debugger told me was actually being passed in and why it was skipping to the throw exception block. 

Was this something that showed up as a result of the auto-app changes, or was this something that showed up as a result of these class changes (from Groovy to Java) in the PR?

**Pull request merged**